src/main/java/com/stripe/model/Source.java

```diff
@@ -0,0 +1,258 @@
+package com.stripe.model;
+
+import com.stripe.exception.APIConnectionException;
+import com.stripe.exception.APIException;
+import com.stripe.exception.AuthenticationException;
+import com.stripe.exception.CardException;
+import com.stripe.exception.InvalidRequestException;
+import com.stripe.net.RequestOptions;
+
+import java.util.Map;
+
+public class Source extends ExternalAccount {
+	Integer amount;
+	String clientSecret;
+	Long created;
+	String currency;
+	String flow;
+	Boolean livemode;
+	SourceOwner owner;
+	String status;
+	String usage;
+	String type;
+
+	// Flow-specific properties
+	SourceReceiverFlow receiver;
+	SourceRedirectFlow redirect;
+	SourceVerificationFlow verification;
+
+	// Type-specific properties
+	Map<String, String> ideal;
+	Map<String, String> sofort;
+	Map<String, String> bancontact;
+	Map<String, String> bitcoin;
+	Map<String, String> achDebit;
+	Map<String, String> sepaDebit;
+
+	public Integer getAmount() {
+		return amount;
+	}
+
+	public void setAmount(Integer amount) {
+		this.amount = amount;
+	}
+
+	public String getClientSecret() {
+		return clientSecret;
+	}
+
+	public void setClientSecret(String clientSecret) {
+		this.clientSecret = clientSecret;
+	}
+
+	public Long getCreated() {
+		return created;
+	}
+
+	public void setCreated(Long created) {
+		this.created = created;
+	}
+
+	public String getCurrency() {
+		return currency;
+	}
+
+	public void setCurrency(String currency) {
+		this.currency = currency;
+	}
+
+	public String getFlow() {
+		return flow;
+	}
+
+	public void setFlow(String flow) {
+		this.flow = flow;
+	}
+
+	public Boolean getLivemode() {
+		return livemode;
+	}
+
+	public void setLivemode(Boolean livemode) {
+		this.livemode = livemode;
+	}
+
+	public SourceOwner getOwner() {
+		return owner;
+	}
+
+	public void setOwner(SourceOwner owner) {
+		this.owner = owner;
+	}
+
+	public String getStatus() {
+		return status;
+	}
+
+	public void setStatus(String status) {
+		this.status = status;
+	}
+
+	public String getUsage() {
+		return usage;
+	}
+
+	public void setUsage(String usage) {
+		this.usage = usage;
+	}
+
+	public String getType() {
+		return type;
+	}
+
+	public void setType(String type) {
+		this.type = type;
+	}
+
+	// Flow-specific getters/setters
+
+	public SourceReceiverFlow getReceiver() {
+		return receiver;
+	}
+
+	public void setReceiver(SourceReceiverFlow receiver) {
+		this.receiver = receiver;
+	}
+
+	public SourceRedirectFlow getRedirect() {
+		return redirect;
+	}
+
+	public void setRedirect(SourceRedirectFlow redirect) {
+		this.redirect = redirect;
+	}
+
+	public SourceVerificationFlow getVerification() {
+		return verification;
+	}
+
+	public void setVerification(SourceVerificationFlow verification) {
+		this.verification = verification;
+	}
+
+	// Type-specific getters/setters
+
+	public Map<String, String> getIdeal() {
```

Comment:
/shrug

I'm really not actually sold on the generic method here. The merchant will need to know what they are dealing with anyway (so they would know what is actually in the HashMap).

Do we want to just have `sepaData`, `idealData`, etc as actual classes that are type-checked? I didn't do this originally because I thought it would be nice to be more generic (so we can add / remove props without cutting a Java release), but it sounds like @brandur doesn't like this idea?